api/app/Actions/Profile/UpdateProfileAction.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Profile;
+
+use App\Exceptions\Profile\ProfileNotFoundException;
+use App\Repositories\User\UserRepository;
+use Illuminate\Auth\AuthenticationException;
+use Illuminate\Support\Facades\Auth;
+
+final class UpdateProfileAction
+{
+    private UserRepository $profileRepository;
+
+    public function __construct(UserRepository $profileRepository)
+    {
+        $this->profileRepository = $profileRepository;
+    }
+
+    public function execute(UpdateProfileRequest $request): UpdateProfileResponse
+    {
+        $user = Auth::user();
+
+        if (!$user) {
+            throw new AuthenticationException();
+        }
```

Comment:
In the current code, there should not be such check because an unauthorized user must not have access to the action at the middleware level.